client/src/components/AdvisorReports/FilterableAdvisorReportsTable.js

```diff
@@ -24,59 +23,52 @@ const GQL_GET_ADVISOR_REPORTS_INSIGHT = gql`
 `
 
 const DEFAULT_WEEKS_AGO = 3
-const OrganizationAdvisorsTableWithLoader = connect(
-  null,
-  mapDispatchToProps
-)(LoaderHOC("isLoading")("data")(OrganizationAdvisorsTable))
-
-class FilterableAdvisorReportsTable extends Component {
-  static propTypes = {
-    showLoading: PropTypes.func.isRequired,
-    hideLoading: PropTypes.func.isRequired
-  }
-
-  constructor() {
-    super()
-    this.state = {
-      filterText: "",
-      export: false,
-      data: [],
-      isLoading: false,
-      selectedData: []
-    }
-    this.handleFilterTextInput = this.handleFilterTextInput.bind(this)
-    this.handleExportButtonClick = this.handleExportButtonClick.bind(this)
-    this.handleRowSelection = this.handleRowSelection.bind(this)
-  }
 
-  componentDidMount() {
-    this.setState({ isLoading: true })
-    this.props.showLoading()
-    API.query(GQL_GET_ADVISOR_REPORTS_INSIGHT).then(data => {
-      this.setState({
-        isLoading: false,
-        data: data.advisorReportInsights
-      })
-      this.props.hideLoading()
-    })
-  }
-
-  handleFilterTextInput(filterText) {
-    this.setState({ filterText: filterText })
+const FilterableAdvisorReportsTable = props => {
+  const [filterText, setFilterText] = useState("")
+  const [selectedData, setSelectedData] = useState([])
```

Comment:
Somehow I was misleaded by the selectedData state, I thought it would be the filteredData. But it actually is the selectedRowsData. Right now if I use the filterText to filter the data and click export to CSV afterwards, without having checked any rows, I get all data in the export. I was expecting it to get only what I see. It is a bit strange, nothing is selected, I get everything back. But if I select rows I only get those ... It is not related to this pull request, maybe we should rethink if this is clear to the users.

Reply:
Yes indeed, this is exactly how it behaves on candidate (I did not change the functionality). Please make a new issue to track this, @maradragan.